action.yml

```diff
@@ -0,0 +1,117 @@
+        # ToDo: Re-enable cache when https://github.com/actions/setup-python/issues/361 is fixed
+        poetry_cache_enabled: 'false'
+
+    - name: Get Image URI
+      shell: bash
+      id: get-image-uri
+      run: |
+        # For testing purposes ( actual value: =master-${{ github.sha }} )
+        imageTag=$(echo ${{ github.ref_name }} | awk '{print tolower($0)}' | sed -e 's|/|-|g')
+        IMAGE_TAGS=$(
+          aws ecr describe-images \
+          --repository-name ${{ inputs.ecr_repository }} \
+          --image-ids imageTag="${imageTag}" \
+          --query 'imageDetails[*].imageTags[0]' \
+          --output json \
+          --region ${{ inputs.aws_region }}
+        )
+        REPOSITORY_URI=$(
+          aws ecr describe-repositories \
+          --repository-name ${{ inputs.ecr_repository }} \
+          --query 'repositories[0].repositoryUri' \
+          --output text \
+          --region ${{ inputs.aws_region }}
+        )
+        IMAGE_URI=$(jq --arg uri "${REPOSITORY_URI}" '.[] | ($uri + ":" + .)' <<< "${IMAGE_TAGS}")
+        echo "image-uri=${IMAGE_URI}" >> $GITHUB_OUTPUT
+
+    - name: Render and Deploy local-exec task definition to Amazon ECS
+      id: deploy-local-task-definition
+      uses: moneymeets/action-ecs-deploy/custom-deploy-steps@feature/MD-7037-implement-ecs-deployment-workflow
+      with:
+        application_id: ${{ inputs.ecr_repository }}-local-exec-${{ inputs.environment }}
+        image_uri: ${{ steps.get-image-uri.outputs.image-uri }}
+        aws_access_key_id: ${{ inputs.aws_access_key_id }}
+        aws_secret_access_key: ${{ inputs.aws_secret_access_key }}
+        aws_region: ${{ inputs.aws_region }}
+
+    - name: Render and Deploy task definition to Amazon ECS
+      id: deploy-task-definition
+      uses: moneymeets/action-ecs-deploy/custom-deploy-steps@feature/MD-7037-implement-ecs-deployment-workflow
+      with:
+        application_id: ${{ inputs.ecr_repository }}-${{ inputs.environment }}
+        image_uri: ${{ steps.get-image-uri.outputs.image-uri }}
+        aws_access_key_id: ${{ inputs.aws_access_key_id }}
+        aws_secret_access_key: ${{ inputs.aws_secret_access_key }}
+        aws_region: ${{ inputs.aws_region }}
+
+    - name: Update service
+      shell: bash
+      run: |
+        aws ecs update-service \
+          --health-check-grace-period-seconds 900 \
+          --task-definition "${{ steps.deploy-task-definition.outputs.latest-task-definition-arn }}" \
+          --cluster ${{ inputs.environment }} \
+          --service ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
+          --desired-count 1 \
+          --region ${{ inputs.aws_region }}
+        
+        aws ecs wait services-stable \
+          --cluster ${{ inputs.environment }} \
+          --service ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
+          --region ${{ inputs.aws_region }}
+
+    - name: Deregister previous local-exec task definition
```

Comment:
Discussed via call; agreed to add condition step for initial deployment; since for initial deployment there wont be a previous task definition to deregister